claat/parser/md/parse.go

```diff
@@ -127,6 +127,14 @@ type stackItem struct {
 	cur *html.Node
 }
 
+func newDocState() *docState {
+	ds := &docState{
+		clab: types.NewCodelab(),
+	}
+
+	return ds
+}
+
```

Comment:
I still don't see the added value of this constructor. It's smelly imo.

This seems to be a wrapper to another constructor, and the result is proceded by more initialization assignments. Feels like either a more generic constructor should be made (not too readable past one variadic arg), or it should be left as a struct init.

And ofc, is also WET.

Reply:
Also nit: do not allocate extra variable. `return &...`

Reply:
A docState struct created by itself is not ready for use. The inner clab field requires init, so docState needs init. A New* function is the pattern for handling init, by producing an object that has all fields zeroed/inited so the caller is not required to do any manual init work on its own.

(handled nit)

Reply:
> The inner clab field requires init, so docState needs init
correct, yet this is all this f(x) is doing, and in 2/3 times it's used it requires some other manual assignment, which makes me think this is an overkill/unnecessary-overhead as opposed to inline struct init (since optional args will be annoying to enforce).

this is more of a comment on the bad design of coupling of everything into a `docState`, but is fine to continue ~

Reply:
It's expected to need to manipulate an object after initializing it. I don't see this as overhead, rather, as reducing overhead. Callers don't need to manually perform identical init steps everywhere, they just call newDocState and it's ready for whatever particular use they have in mind.

Reply:
seems fine then